Sources/Components/Radio/Radio.swift

```diff
@@ -0,0 +1,136 @@
+import SwiftUI
+
+extension Warp {
+    /// Radio buttons allow users to select one option from a set.
+    ///
+    /// `Radio` is a customizable component that allows users
+    /// to create a list of radio buttons where only one can be selected at any given time.
+    ///
+    /// - Parameters:
+    ///   - selectedOption: A binding to the currently selected option.
+    ///   - options: An array of options that conform to `Identifiable` and `Hashable`.
+    ///   - label: A closure that provides a label for each option.
+    ///   - state: The state of the radio button group (default, error, disabled).
+    public struct Radio<Option: Identifiable & Hashable>: View {
+        /// A binding to the currently selected option.
+        @Binding var selectedOption: Option
+        /// An array of options that conform to `Identifiable` and `Hashable`.
+        var options: [Option]
+        /// A closure that provides a label for each option.
+        var label: (Option) -> String
+        /// The state of the radio button group (default, error, disabled).
+        var state: RadioButtonState
+        /// Object that will provide needed colors.
+        private let colorProvider: ColorProvider = Warp.Color
+
+        /// Initializes a new `Radio`.
+        ///
+        /// - Parameters:
+        ///   - selectedOption: A binding to the currently selected option.
+        ///   - options: An array of options that conform to `Identifiable` and `Hashable`.
+        ///   - label: A closure that provides a label for each option.
+        ///   - state: The state of the radio button group (default, error, disabled).
+        public init(selectedOption: Binding<Option>,
+                    options: [Option],
+                    label: @escaping (Option) -> String,
+                    state: RadioButtonState = .default) {
+            self._selectedOption = selectedOption
+            self.options = options
+            self.label = label
+            self.state = state
+        }
+        
+        public var body: some View {
+            VStack(alignment: .leading, spacing: Spacing.spacing200) {
+                ForEach(options) { option in
+                    RadioButton(isSelected: selectedOption == option,
+                                label: label(option),
+                                state: state) {
+                        if state != .disabled {
+                            selectedOption = option
+                        }
```

Comment:
You seem to have the same check inside the `RadioButton` itself:

```swift
.onTapGesture {
    if state != .disabled {
        action()
    }
}
```

Is that intentional?

Reply:
Good point, I am gonna remove it from the group